crates/miden-objects/src/block/signed_block.rs

```diff
@@ -0,0 +1,77 @@
+use core::ops::{Deref, DerefMut};
+
+use miden_core::utils::{
+    ByteReader,
+    ByteWriter,
+    Deserializable,
+    DeserializationError,
+    Serializable,
+};
+use miden_crypto::dsa::ecdsa_k256_keccak::{PublicKey, Signature};
+
+use crate::block::{BlockHeader, ProposedBlock};
+
+#[derive(Debug, Clone)]
+pub struct SignedBlock {
+    header: BlockHeader,
+    proposed_block: ProposedBlock,
+    signature: Signature,
+}
```

Comment:
I thought this would be more like:
```rust
pub struct SignedBlock {
    header: BlockHeader, // this will contain the signature
    body: BlockBody,
}
```
What's the reason to have proposed block here?